src/System.Management.Automation/utils/CryptoUtils.cs

```diff
@@ -2,440 +2,264 @@
+                // DWORD bitlen
+                int bitLen = ToInt32LE(blob, offset + 12);
+
+                // DWORD public exponent
+                RSAParameters rsap = new RSAParameters();
+                rsap.Exponent = new byte[3];
+                rsap.Exponent[0] = blob[offset + 18];
+                rsap.Exponent[1] = blob[offset + 17];
+                rsap.Exponent[2] = blob[offset + 16];
+            
+                int pos = offset + 20;
+                int byteLen = (bitLen >> 3);
+                rsap.Modulus = new byte[byteLen];
+                Buffer.BlockCopy(blob, pos, rsap.Modulus, 0, byteLen);
+                Array.Reverse(rsap.Modulus);
+
+                return rsap;
+            } 
+            catch (Exception ex) 
+            {
+                throw new CryptographicException("Invalid public key blob", ex);
+            }
+        }
 
-        /// <summary>
-        /// RSA Key.
-        /// </summary>
-        public const int CALG_RSA_KEYX =
-            (PSCryptoNativeUtils.ALG_CLASS_KEY_EXCHANGE |
-            (PSCryptoNativeUtils.ALG_TYPE_RSA | PSCryptoNativeUtils.ALG_SID_RSA_ANY));
+        internal static byte[] ToCapiPublicKeyBlob(RSA rsa) 
+        {
+            if (rsa == null)
+            {
+                throw new ArgumentNullException("rsa");
+            }
 
-        /// <summary>
-        /// Create a key for encryption.
-        /// </summary>
-        public const int ALG_CLASS_KEY_EXCHANGE = (5) << (13);
+            RSAParameters p = rsa.ExportParameters(false);
+            int keyLength = p.Modulus.Length; // in bytes
+            byte[] blob = new byte[PUBLICKEYBLOB_HEADER_LEN + keyLength];
+
+            blob[0] = (byte)PUBLICKEYBLOB;    // Type - PUBLICKEYBLOB (0x06)
+            blob[1] = (byte)CUR_BLOB_VERSION; // Version - Always CUR_BLOB_VERSION (0x02)
+            // [2], [3]                 // RESERVED - Always 0
+            blob[5] = (byte)CALG_RSA_KEYX;    // ALGID - Always 00 a4 00 00 (for CALG_RSA_KEYX)
+            blob[8] = 0x52;             // Magic - RSA1 (ASCII in hex)
+            blob[9] = 0x53;
+            blob[10] = 0x41;
+            blob[11] = 0x31;
+
+            byte[] bitlen = GetBytesLE(keyLength << 3);
+            blob[12] = bitlen[0];       // bitlen
+            blob[13] = bitlen[1];
+            blob[14] = bitlen[2];
+            blob[15] = bitlen[3];
+
+            // public exponent (DWORD)
+            int pos = 16;
+            int n = p.Exponent.Length;
+            while (n > 0)
+            {
+                blob[pos++] = p.Exponent[--n];
+            }
 
-        /// <summary>
-        /// Create a RSA key pair.
-        /// </summary>
-        public const int ALG_TYPE_RSA = (2) << (9);
+            // modulus
+            pos = 20;
+            byte[] part = p.Modulus;
+            int len = part.Length;
+            Array.Reverse(part, 0, len);
+            Buffer.BlockCopy(part, 0, blob, pos, len);
```

Comment:
I think this would be more readable if `part` name was `modulus`.  Also `len` is not needed since it is the same as `keyLength` above.
```
Buffer.BlockCopy(modulus, 0, blob, pos, keyLength);
```